### ISSUE TITLE
Doesn't assign existing S3::Client from Aws::S3::Encryption::Client during init

### ISSUE THREAD

**Issue by kmonihen** [stelligent/crossing#20]
During init, if an S3:Client object is not passed in, a new S3:Client object is created rather than being assigned from the Aws::S3::Encryption::Client object. This means that none of the attribute values from the encrypted client are passed to the unencrypted client.

**Comment by kmonihen** [comment]
This code:

 ```
kms = Aws::KMS::Client.new region: ENV['region']
  s3 = Aws::S3::Encryption::Client.new(
    kms_key_id: '...',
    kms_client: kms,
    region: 'us-east-1'
  )
  crossing = Crossing.new(s3)
```

Raises an error because the new S3:Client object does not have the region set:
> Aws::Errors::MissingRegionError: missing region; use :region option or export region name to ENV['AWS_REGION']

**Comment by kmonihen** [comment]
Fixed in 0.1.2 build.
